### ISSUE TITLE
Widespread "Optimizing assemblies for size is not supported for the selected publish configuration" failures

### ISSUE THREAD

**Issue by EgorBo** [dotnet/runtime#95712]
Looks like CI is in a bad state, e.g. https://github.com/dotnet/runtime/pull/95709/checks?check_run_id=19389558137

### Known Issue Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

```json
{
  "ErrorMessage": "Optimizing assemblies for size is not supported for the selected publish configuration",
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

 ### Known issue validation
**Build: :mag_right:** 
**Result validation: :warning:** Validation could not be done without an Azure DevOps build URL on the issue. Please add it to the "**Build: :mag_right:**" line.
**Validation performed at:** 12/7/2023 12:25:05 AM UTC

### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[2330788](https://dev.azure.com/dnceng/internal/_build/results?buildId=2330788)|dotnet-runtime|SuperPMI collect smoke_tests nativeaot windows arm64 checked|[Log](https://dev.azure.com/dnceng/7ea9116e-9fac-403d-b258-b31fcf1bb293/_apis/build/builds/2330788/logs/3332)||
|[491070](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491070)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/491070/logs/1235)|dotnet/runtime#94109|

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[491374](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491374)|dotnet/runtime|[iOS.Device.ExportManagedSymbols.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491374&view=ms.vss-test-web.build-test-results-tab&runId=11381708&resultId=100002)|dotnet/runtime#95715|
|[491404](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491404)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491404&view=ms.vss-test-web.build-test-results-tab&runId=11380680&resultId=100001)|dotnet/runtime#95719|
|[491346](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491346)|dotnet/runtime|[iOS.Device.ExportManagedSymbols.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491346&view=ms.vss-test-web.build-test-results-tab&runId=11379354&resultId=100002)|dotnet/runtime#95711|
|[491351](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491351)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491351&view=ms.vss-test-web.build-test-results-tab&runId=11379360&resultId=100001)|dotnet/runtime#95714|
|[491343](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491343)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491343&view=ms.vss-test-web.build-test-results-tab&runId=11379358&resultId=100001)|dotnet/runtime#94737|
|[491290](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491290)|dotnet/runtime|[iOS.Device.ExportManagedSymbols.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491290&view=ms.vss-test-web.build-test-results-tab&runId=11378658&resultId=100002)|dotnet/runtime#95701|
|[491268](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491268)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491268&view=ms.vss-test-web.build-test-results-tab&runId=11378214&resultId=100001)|dotnet/runtime#95670|
|[491247](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491247)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491247&view=ms.vss-test-web.build-test-results-tab&runId=11377918&resultId=100001)|dotnet/runtime#95139|
|[491230](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491230)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491230&view=ms.vss-test-web.build-test-results-tab&runId=11377742&resultId=100001)|dotnet/runtime#95711|
|[491150](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491150)|dotnet/runtime|[iOS.Device.ExportManagedSymbols.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491150&view=ms.vss-test-web.build-test-results-tab&runId=11375480&resultId=100002)|dotnet/runtime#95710|
|[491139](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491139)|dotnet/runtime|[iOS.Device.ExportManagedSymbols.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491139&view=ms.vss-test-web.build-test-results-tab&runId=11374818&resultId=100002)|dotnet/runtime#94446|
|[491132](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491132)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491132&view=ms.vss-test-web.build-test-results-tab&runId=11374350&resultId=100001)|dotnet/runtime#95493|
|[491092](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491092)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491092&view=ms.vss-test-web.build-test-results-tab&runId=11373134&resultId=100001)|dotnet/runtime#91229|
|[491100](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491100)|dotnet/runtime|[iOS.Device.ExportManagedSymbols.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491100&view=ms.vss-test-web.build-test-results-tab&runId=11373132&resultId=100002)|dotnet/runtime#95530|
|[491077](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491077)|dotnet/runtime|[iOS.Device.ExportManagedSymbols.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491077&view=ms.vss-test-web.build-test-results-tab&runId=11372376&resultId=100002)|dotnet/runtime#95698|
|[491065](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491065)|dotnet/runtime|[iOS.Device.ExportManagedSymbols.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491065&view=ms.vss-test-web.build-test-results-tab&runId=11372250&resultId=100002)|dotnet/runtime#95707|
|[491054](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491054)|dotnet/runtime|[tvOS.Device.Aot.Test.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=491054&view=ms.vss-test-web.build-test-results-tab&runId=11371440&resultId=100001)|dotnet/runtime#95709|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|19|19|19|

**Comment by steveisok** [comment]
Weird how it's only a small number of hits. Diving into the PR's show much more.

**Comment by jkotas** [comment]
Likely introduced by #95712

**Comment by MichalStrehovsky** [comment]
Fix in #95718
